libwacom/libwacom-database.c

```diff
@@ -279,8 +323,81 @@ parse_stylus_id(const char *str, WacomStylusId *id)
 	return ret;
 }
 
+static char *
+string_or_fallback (GKeyFile *keyfile, const char *group,
+		    const char *key, const char *fallback, GError *error)
+{
+	GError *local_error = NULL;
+	bool ret;
+
+	ret = g_key_file_has_key(keyfile, group, key, &local_error);
+	if (local_error) {
+		g_warning ("String fallback error: %s\n", local_error->message);
+		g_clear_error (&local_error);
+	}
+	if (ret)
+		return g_key_file_get_string(keyfile, group, key, &error);
+
+	return g_strdup(fallback);
+}
+
+static int
+int_or_fallback (GKeyFile *keyfile, const char *group,
+		 const char *key, int fallback, GError *error)
+{
+	GError *local_error = NULL;
+	bool ret;
+
+	ret = g_key_file_has_key(keyfile, group, key, &local_error);
+	if (local_error) {
+		g_warning ("int fallback error: %s\n", local_error->message);
+		g_clear_error (&local_error);
+	}
+
+	if (ret)
+		return g_key_file_get_integer(keyfile, group, key, &error);
+
+	return fallback;
+}
+
+static bool
+boolean_or_fallback (GKeyFile *keyfile, const char *group,
+		     const char *key, bool fallback, WacomStylus *stylus, GError *error)
+{
+	GError *local_error = NULL;
+	bool ret;
+
+	ret = g_key_file_has_key(keyfile, group, key, &local_error);
+	if (local_error) {
+		g_warning ("boolean fallback error: %s\n", local_error->message);
+		g_clear_error (&local_error);
+	}
+
+	if (ret)
+		return g_key_file_get_boolean(keyfile, group, key, &error);
+
+	return fallback;
+}
+
+static gchar **
+garray_to_string_list(GArray *array)
```

Comment:
this sounds like it's a generic array->string list conversion but it's a `WacomStylusId` -> hex string converstion, let's rename this please, e.g. 

```
stylus_ids_as_hex(GArray *array)
```
or something similar